qhub/initialize.py

```diff
@@ -337,63 +344,45 @@ def render_config(
-            # to be dynamically filled since digital ocean updates the
-            # versions so frequently
-            config["digital_ocean"][
-                "kubernetes_version"
-            ] = digital_ocean.kubernetes_versions()[0]["slug"]
+        config["digital_ocean"] = DIGITAL_OCEAN.copy()
+        _set_kubernetes_version(config, kubernetes_version, cloud_provider)
+
     elif cloud_provider == "gcp":
         config["theme"]["jupyterhub"][
             "hub_subtitle"
         ] = "Autoscaling Compute Environment on Google Cloud Platform"
-        config["google_cloud_platform"] = GOOGLE_PLATFORM
-        if kubernetes_version:
-            config["google_cloud_platform"]["kubernetes_version"] = kubernetes_version
+        config["google_cloud_platform"] = GOOGLE_PLATFORM.copy()
+        _set_kubernetes_version(config, kubernetes_version, cloud_provider)
 
         if "PROJECT_ID" in os.environ:
             config["google_cloud_platform"]["project"] = os.environ["PROJECT_ID"]
         elif not disable_prompt:
             config["google_cloud_platform"]["project"] = input(
                 "Enter Google Cloud Platform Project ID: "
             )
+
     elif cloud_provider == "azure":
         config["theme"]["jupyterhub"][
             "hub_subtitle"
         ] = "Autoscaling Compute Environment on Azure"
-        config["azure"] = AZURE
-
-        if kubernetes_version:
-            config["azure"]["kubernetes_version"] = kubernetes_version
-        else:
-            # first kubernetes version returned by azure sdk is
-            # the newest version of kubernetes supported this field needs
-            # to be dynamically filled since azure updates the
-            # versions so frequently
-            config["azure"]["kubernetes_version"] = azure_cloud.kubernetes_versions(
-                config["azure"]["region"]
-            )[0]
+        config["azure"] = AZURE.copy()
+        _set_kubernetes_version(config, kubernetes_version, cloud_provider)
 
     elif cloud_provider == "aws":
         config["theme"]["jupyterhub"][
             "hub_subtitle"
         ] = "Autoscaling Compute Environment on Amazon Web Services"
-        config["amazon_web_services"] = AMAZON_WEB_SERVICES
-        if kubernetes_version:
-            config["amazon_web_services"]["kubernetes_version"] = kubernetes_version
+        config["amazon_web_services"] = AMAZON_WEB_SERVICES.copy()
+        _set_kubernetes_version(config, kubernetes_version, cloud_provider)
+
     elif cloud_provider == "local":
         config["theme"]["jupyterhub"][
             "hub_subtitle"
         ] = "Autoscaling Compute Environment"
-        config["local"] = LOCAL
+        config["local"] = LOCAL.copy()
 
-    config["profiles"] = DEFAULT_PROFILES
-    config["environments"] = DEFAULT_ENVIRONMENTS
+    config["profiles"] = DEFAULT_PROFILES.copy()
+    config["environments"] = DEFAULT_ENVIRONMENTS.copy()
 
```

Comment:
Is it possible to add the Kubernetes version checker here?
```python
### Verify Kubernetes version
_set_kubernetes_version(config, kubernetes_version, cloud_provider)
```
Then we could have just one call for it, and remove it from the conditionals above

Reply:
@viniciusdc the current state requires that `cloud_provider` be passed to the `_set_kubernetes_version` which means we can't simply replace all of these functions calls with one call. We might be able to do something like:
```python
for cloud_provider in ["aws", "do", "azure", "gcp", "local"]:
    _set_kubernetes_version(config, kubernetes, cloud_provider)
```